core-bundle/src/Doctrine/Backup/Dumper.php

```diff
@@ -78,46 +96,61 @@ private function dumpData(Connection $connection, Table $table): \Generator
     {
         yield sprintf('-- BEGIN DATA %s', $table->getName());
         $values = [];
+        $columnBindingTypes = [];
+        $columnCharsets = [];
 
         foreach ($table->getColumns() as $column) {
-            $values[] = sprintf('`%s` AS `%s`', $column->getName(), $column->getName());
+            $columnName = $column->getName();
+            $values[] = "`$columnName` AS `$columnName`";
+            $columnBindingTypes[$columnName] = $column->getType()->getBindingType();
+            $columnCharsets[$columnName] = strtolower($column->getPlatformOptions()['charset'] ?? '');
         }
 
-        $rows = $connection->executeQuery(sprintf('SELECT %s FROM `%s`', implode(', ', $values), $table->getName()));
+        $values = implode(', ', $values);
+        $tableName = $table->getName();
+        $rows = $connection->executeQuery("SELECT $values FROM `$tableName`");
 
         foreach ($rows->iterateAssociative() as $row) {
             $insertColumns = [];
             $insertValues = [];
 
             foreach ($row as $columnName => $value) {
-                $insertColumns[] = sprintf('`%s`', $columnName);
-                $insertValues[] = $this->formatValueForDump($value, $table->getColumn($columnName), $connection);
+                $insertColumns[] = "`$columnName`";
+                $insertValues[] = $this->formatValueForDump(
+                    $value,
+                    $columnBindingTypes[$columnName],
+                    $columnCharsets[$columnName],
+                    $connection
+                );
             }
 
-            yield sprintf(
-                'INSERT INTO `%s` (%s) VALUES (%s);',
-                $table->getName(),
-                implode(', ', $insertColumns),
-                implode(', ', $insertValues)
-            );
+            $insertColumns = implode(', ', $insertColumns);
+            $insertValues = implode(', ', $insertValues);
+
+            yield "INSERT INTO `$tableName` ($insertColumns) VALUES ($insertValues);";
         }
     }
 
-    /**
-     * @param mixed $value
-     */
-    private function formatValueForDump($value, Column $column, Connection $connection): string
+    private function formatValueForDump(?string $value, int $columnBindingType, string $columnCharset, Connection $connection): string
     {
         if (null === $value) {
             return 'NULL';
         }
 
+        if ('' === $value) {
+            return '';
```

Comment:
```suggestion
            return "''";
```
Should this be something like this?

Reply:
yes, @ausi is correct

Reply:
Of course, I've fixed and covered with tests in https://github.com/contao/contao/pull/4228/commits/962955401ef9361f518afdc089a6e0af1975a534.